DotNet/Census/Application/Services/ScheduleService.cs

```diff
@@ -39,23 +39,30 @@ public ScheduleService(
 
     public override async Task StartAsync(CancellationToken cancellationToken)
     {
-        Scheduler = await _schedulerFactory.GetScheduler(cancellationToken);
+        try
+        {
+            Scheduler = await _schedulerFactory.GetScheduler(cancellationToken);
+
+            Scheduler.JobFactory = _jobFactory;
 
-        Scheduler.JobFactory = _jobFactory;
+            var configRepo = _serviceScopeFactory.CreateScope().ServiceProvider.GetRequiredService<ICensusConfigManager>();
 
-        var configRepo = _serviceScopeFactory.CreateScope().ServiceProvider.GetRequiredService<ICensusConfigManager>();
+            List<CensusConfigEntity> facilities = (await configRepo.GetAllFacilities(cancellationToken)).ToList();
 
-        List<CensusConfigEntity> facilities = (await configRepo.GetAllFacilities(cancellationToken)).ToList();
+            using var censusSchedulingRepo = _serviceScopeFactory.CreateScope().ServiceProvider.GetRequiredService<ICensusSchedulingRepository>();
 
-        using var censusSchedulingRepo = _serviceScopeFactory.CreateScope().ServiceProvider.GetRequiredService<ICensusSchedulingRepository>();
+            foreach (CensusConfigEntity facility in facilities)
+            {
+                censusSchedulingRepo.CreateJobAndTrigger(facility, Scheduler);
+            }
 
-        foreach (CensusConfigEntity facility in facilities)
+            await Scheduler.Start(cancellationToken);
+            _logger.LogInformation("Scheduler started.");
+        }
+        catch (Exception ex)
         {
-            censusSchedulingRepo.CreateJobAndTrigger(facility, Scheduler);
+            _logger.LogError(ex, "Something went wrong scheduling a Census job: {1}.", ex.Message);
```

Comment:
If an error occurs, that will still prevent the service to restart. Don't we want to log that particular error (maybe even audit it), but still let the service start?